File: src/main/java/spaceEngineers/SeBlock.java
package spaceEngineers;

import eu.iv4xr.framework.spatial.Vec3;

public class SeBlock extends SeEntity {
    public float maxIntegrity;
    public float buildIntegrity;
    public float integrity;
    public String blockType;

    public Vec3 minPosition;
    public Vec3 maxPosition;
    public Vec3 size;
    public Vec3 orientationForward;
    public Vec3 orientationUp;
}

File: src/commonMain/kotlin/spaceEngineers/model/SeBlock.kt
package spaceEngineers.model


data class SeBlock(
    override val id: String,
    override val position: Vec3,
    val blockType: String
) : SeEntity {
    var maxIntegrity = 0f
    var buildIntegrity = 0f
    var integrity = 0f

    var minPosition: Vec3? = null
    var maxPosition: Vec3? = null
    var size: Vec3? = null
    var orientationForward: Vec3? = null
    var orientationUp: Vec3? = null
}